library/x509_crl.c

```diff
@@ -295,7 +295,7 @@ int x509_crl_parse_der( x509_crl *chain,
     /*
      * Copy raw DER-encoded CRL
      */
-    if( ( p = polarssl_malloc( buflen ) ) == NULL )
+    if( buflen != 0 && ( ( p = polarssl_malloc( buflen ) ) == NULL ) )
```

Comment:
The check for `buflen == 0` should be made earlier. See https://github.com/ARMmbed/mbedtls/pull/1009/files#r153007520